dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,63 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(**kwargs):
+    """Split args on those applicable to Pipeline constructor and the rest."""
+    fca = inspect.getfullargspec(Pipeline.__init__)  # Full Ctor Args
+    ctor_args = dict(filter(lambda x: x[0] in fca.args or x[0] in fca.kwonlyargs, kwargs.items()))
```

Comment:
This should be the other way round: detect arguments for this function - if not found, try Pipeline - if not found either, raise an error.
Rationale: if we add new arguments to Pipeline's constructor, we may break users' code because a named argument would be redirected to the Pipeline constructor instead.

Reply:
Done